grammar.js

```diff
@@ -1178,18 +1178,27 @@ module.exports = grammar(C, {
       $.qualified_identifier,
     ),
 
-    assignment_expression: ($, original) => choice(
-      original,
-      prec.right(PREC.ASSIGNMENT, seq(
-        field('left', $._assignment_left_expression),
-        field('operator', choice(
-          'and_eq',
-          'or_eq',
-          'xor_eq',
-        )),
-        field('right', $._expression),
+    assignment_expression: $ => prec.right(PREC.ASSIGNMENT, seq(
+      field('left', $._assignment_left_expression),
+      field('operator', choice(
+        '=',
```

Comment:
Why do we need to move the c operators here instead of inheriting

Reply:
rhs includes initializer list

Reply:
I see, this is only supported in c++, so we have to copy the list of c operators to the assignment_expression of c++ grammar?

Reply:
yep

Reply:
Btw, Amaan, I hit another problem related to assignment_expression, right now c++ parser is not able to parse a parenthesized assignment_expression with the c operators, for example, if(arg += 1) cannot be parsed by c++ parser, but if(arg and_eq 1) can. Have you ever hit the same problem?

Reply:
yeah the last tidbit in my pr message is just about that, it's a complicated one that I'm not sure I can solve but I'll try some more ideas tonight

Reply:
This is great to know, my project is confronted with this issue, I am glad to help if you need some hands, or someone to review, I am looking forward to your fix, good luck!